Add sending allowed children notification article

erikjanwestendorp opened the pull request:
During [DF22](https://df22.nl/) @dawoe did a talk of the unknown features of Umbraco. One of the subjecs was `SendingAllowedChildrenNotification`

I think it would be nice if we can add an example to the docs

sofietoft commented on Reference/Notifications/SendingAllowedChildrenNotification/index.md:
```diff
@@ -0,0 +1,67 @@
+---
+
+# Sending Allowed Children Notification
+The `SendingAllowedChildrenNotification` enable you to manipulate the content types that will be shown in the create menu.
```
```suggestion

The `SendingAllowedChildrenNotification` enables you to manipulate the content types that will be shown in the create menu when adding new content in the backoffice.
```

sofietoft commented on Reference/Notifications/SendingAllowedChildrenNotification/index.md:
```diff
@@ -0,0 +1,67 @@
+The `SendingAllowedChildrenNotification` enable you to manipulate the content types that will be shown in the create menu.
+
+## Usage
+Example usage of the `SendingAllowedChildrenNotification` - for example show only the settings content type if none have been added yet.
```
How about rephrasing this part to:

```suggestion

With the example below we can ensure that a content type cannot be selected if the type already exists in the Content tree.
```

sofietoft reviewed (commented):
Thanks for the PR @! 💪 

I think this is a great addition to our docs.
I've added a couple of comments.

Finally, is it just me, or is it a little confusing that `contentId` is used as both the placeholder for the ID of the content type AND as the variable holding the info? 🤔 
Or perhaps I'm reading that all wrong? 😅

erikjanwestendorp commented:
@sofietoft Thanks for the review, just updated the PR with your suggestions! I think you are right that it could be a little bit confusing so I updated the code sample. (added a contentIdKey variable) 😄

sofietoft commented:
Excellent! Thanks for applying the changes @erikjanwestendorp ! 👏 

I'll just run it by our development team for a final checkup - even though I'm sure it all checks out!

erikjanwestendorp commented:
@sofietoft thanks! Would it be an idea to add this to the [Umbraco.Docs.Samples](https://github.com/umbraco/Umbraco.Docs.Samples) repo as well? 😄

sofietoft commented:
YES! Very much so 👏 
This would be excellent for the Samples project as well!

elit0451 commented on Reference/Notifications/SendingAllowedChildrenNotification/index.md:
````diff
@@ -0,0 +1,69 @@
+        }
+    }
+}
+```
````
After the code example you can mention:

> You also need to register this notification handler, like:
> 
> ```csharp
> .AddNotificationHandler<SendingAllowedChildrenNotification, SendingAllowedChildrenNotificationHandler>()
>                 
> ```

elit0451 commented on Reference/Notifications/SendingAllowedChildrenNotification/index.md:
```diff
@@ -0,0 +1,69 @@
+                // Get all children of current page
+                var childNodes = content.ChildrenForAllCultures.ToList();
+
+                // If there is a Settings page already created, then don't allow it anymore
```
In case you want to show the `Settings.ModelTypeAlias` example you had:

```suggestion
                // If there is a Settings page already created, then don't allow it anymore
                // You can also use the ModelTypeAlias property from your PublishedModel for comparison,
                // like Settings.ModelTypeAlias if you have set models builder to generate SourceCode models
```

elit0451 reviewed (commented):
Thanks for documenting the usage of this notification @erikjanwestendorp 🙌 . Overall it looks good, I just pushed a few minor changes:

- changed "content type" to "document type" so we are consistent with the names in the backoffice;
- changed the `Settings.ModelTypeAlias` to `"settings"` because for the first one to work, you'd have to set the ModelsBuilder mode to SourceCode, so instead we can mention this in a comment, like my suggestion.

It might be nice to mention by the end of the article that they need to register this notification handler, so I also suggested this change.

erikjanwestendorp commented:
@elit0451 Thanks for your suggestions. Just updated the PR with your suggestions.

elit0451 commented on Reference/Notifications/SendingAllowedChildrenNotification/index.md:
````diff
@@ -0,0 +1,93 @@
+}
+```
+
+For more information av
+
+:::note
+The license for Umbraco Deploy comes with a recurring yearly fee. Learn more about this and pricing on [Umbraco.com](https://umbraco.com/products/umbraco-deploy/).
+:::
````
I believe this is a wrong paste and perhaps you want to refer to https://our.umbraco.com/documentation/Reference/Notifications/#registering-notification-handlers instead 😉

erikjanwestendorp replied:
oopss... 🙈 my bad. Just updated 😄

elit0451 commented on Reference/Notifications/SendingAllowedChildrenNotification/index.md:
````diff
@@ -0,0 +1,93 @@
+
+```csharp
+public void ConfigureServices(IServiceCollection services)
+{
+#pragma warning disable IDE0022 // Use expression body for methods
+    services.AddUmbraco(_env, _config)
+        .AddBackOffice()             
+        .AddWebsite()
+        .AddComposers()
+        .AddNotificationHandler<SendingAllowedChildrenNotification, SendingAllowedChildrenNotificationHandler>()
+        .Build();
+#pragma warning restore IDE0022 // Use expression body for methods
+}
````
```suggestion
{
    services.AddUmbraco(_env, _config)
        .AddBackOffice()             
        .AddWebsite()
        .AddComposers()
        .AddNotificationHandler<SendingAllowedChildrenNotification, SendingAllowedChildrenNotificationHandler>()
        .Build();
}
```

sofietoft commented:
Looks like we're all good to go here 😄 
Thanks again for the PR @erikjanwestendorp , and thank you for reviewing @elit0451 🙌 

I'll merge it in!
